tokio-fs/src/lib.rs

```diff
@@ -0,0 +1,64 @@
+//! Asynchronous filesystem manipulation operations (and stdin, stdout, stderr).
+//!
+//! This module contains basic methods and types for manipulating the contents
+//! of the local filesystem from within the context of the Tokio runtime.
+//!
+//! Tasks running on the Tokio runtime are expected to be asynchronous, i.e.,
+//! they will not block the thread of execution. Filesystem operations do not
+//! satisfy this requirement. In order to perform filesystem operations
+//! asynchronously, this library uses the [`blocking`][blocking] annotation
+//! to signal to the runtime that a blocking operation is being performed. This
+//! allows the runtime to compensate.
+//!
+//! [blocking]: https://docs.rs/tokio-threadpool/0.1/tokio_threadpool/fn.blocking.html
+
+#[macro_use]
+extern crate futures;
+extern crate tokio_io;
+extern crate tokio_threadpool;
+
+pub mod file;
+mod stdin;
+mod stdout;
+mod stderr;
+
+pub use file::File;
+pub use stdin::{stdin, Stdin};
+pub use stdout::{stdout, Stdout};
+pub use stderr::{stderr, Stderr};
```

Comment:
I don't disagree, but it seemed most appropriate in this crate. The crate could be named `tokio-blocking-io` but that also is weird... Filesystem ops are the primary example of blocking io, so I just named it `tokio-fs`.

It could be that it is worth splitting this logic out.